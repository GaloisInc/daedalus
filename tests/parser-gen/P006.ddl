def NT1 = @Match1 ('0'..'7')
def NT2 = { Match1 '(' ; Match1 '{' ; Match1 '}' ; Match1 ')' }
def Main = { x = Many (4..5) NT1 ; y = Many 2 NT2 }
