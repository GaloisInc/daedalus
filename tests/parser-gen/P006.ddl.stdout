module P006
 
--- Imports:
 
--- Type defs:
type P006.Main = { x: [{}]
                 ; y: [uint 8]
                 }
 
--- Rules:
 
P006.NT1 : Grammar {} =
  @Match ('0' .. '7')
 
P006.NT2 : Grammar (uint 8) =
  do @Match {'('}
     @Match {'{'}
     @Match {'}'}
     ($$ : uint 8) <- Match {')'}
     pure $$
 
P006.Main : Grammar P006.Main =
  do (x : [{}]) <- Many[ 4 .. 5 ] P006.NT1
     (y : [uint 8]) <- Many[2] P006.NT2
     pure {x = x,
           y = y}
 
P006.NT1_ : Grammar {} =
  @Match ('0' .. '7')
 
P006.NT2_ : Grammar {} =
  do @Match {'('}
     @Match {'{'}
     @Match {'}'}
     @Match {')'}
 
P006.Main_ : Grammar {} =
  do @Many[ 4 .. 5 ] P006.NT1_
     @Many[2] P006.NT2_
